File: ts/project/.sdk/src/cmp/swift/fragment/EntityListOp.fragment.swift
// List operation fragment (spliced into the entity class by
// EntityOperation_swift; only the EJECT region is emitted).

import Foundation

final class EntyClassListOpFragment: EntyClass {
// EJECT-START
public override func list(_ reqmatch: VMap?, _ ctrl: VMap?) throws -> Value {
  var ctxmap: [String: Any?] = ["opname": "list", "match": match, "data": data]
  if let ctrl = ctrl { ctxmap["ctrl"] = ctrl }
  if let reqmatch = reqmatch { ctxmap["reqmatch"] = reqmatch }
  let ctx = utility.makeContext(ctxmap, entctx)

  return try runOp(ctx) {
    if let result = ctx.result {
      if let rm = result.resmatch { self.match = rm }
    }
  }
}
// EJECT-END
}

File: ts/project/.sdk/src/cmp/csharp/fragment/EntityListOp.fragment.cs
// List operation fragment (spliced into the entity class by
// EntityOperation_csharp; only the EJECT region is emitted).

namespace ProjectNameSdk.Entity;

public class EntyClassListOpFragment : EntyClass
{
    public EntyClassListOpFragment(ProjectNameSDK client) : base(client) { }

// EJECT-START
public override object? List(Dictionary<string, object?>? reqmatch,
    Dictionary<string, object?>? ctrl = null)
{
    var ctx = utility.MakeContext(new Dictionary<string, object?>
    {
        ["opname"] = "list",
        ["ctrl"] = ctrl,
        ["match"] = match,
        ["data"] = data,
        ["reqmatch"] = reqmatch,
    }, entctx);

    return RunOp(ctx, () =>
    {
        if (ctx.Result != null)
        {
            if (ctx.Result.Resmatch != null)
            {
                match = ctx.Result.Resmatch;
            }
        }
    });
}
// EJECT-END
}
